File: src/test/java/calculator/DelimiterTest.java
package calculator;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.assertj.core.api.Assertions.assertThat;

class DelimiterTest {

    @DisplayName(value = "기본 구분자를 기반으로 문자열 분리")
    @ParameterizedTest
    @ValueSource(strings = {"1,2:3"})
    void splitBasicDelimiter(final String text) {
        assertThat(Delimiter.split(text)).isEqualTo(new String[]{"1", "2", "3"});
    }

    @DisplayName(value = "커스텀 구분자를 기반으로 문자열 분리")
    @ParameterizedTest
    @ValueSource(strings = {"//?\n1?2?3"})
    void splitCustomDelimiter(final String text) {
        assertThat(Delimiter.split(text)).isEqualTo(new String[]{"1", "2", "3"});
    }

    @DisplayName(value = "커스텀 구분자 찾기")
    @ParameterizedTest
    @ValueSource(strings = {"//?\n1?2?3"})
    void findCustomDelimiter(final String text) {
        assertThat(Delimiter.customize(text)).isEqualTo("?");
    }

}

File: src/test/kotlin/calculator/DelimiterTest.kt
package calculator

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource
import org.junit.jupiter.params.provider.ValueSource

class DelimiterTest {

    @ParameterizedTest
    @ValueSource(strings = ["0,1,2", "1,2,3"])
    fun `기본 구분자로 생성 성공`(target: String) {
        assertThat(Delimiter(target).regex.toString()).isEqualTo("[,:]")
    }

    @ParameterizedTest
    @CsvSource("""//;\n0;2;3|;""", """//;\n1;0;3|;""", """//.\n1.2.3|.""", delimiter = '|')
    fun `커스텀 구분자로 생성 성공`(target: String, expected: String) {
        assertThat(Delimiter(target).regex.toString()).isEqualTo(Regex.escape(expected))
    }
}
